src/keepass_fmt_plug.c

```diff
@@ -158,103 +288,135 @@ static int crypt_all(int *pcount, struct db_salt *salt)
-			unsigned char dec_buf[32];
+			else if (cur_salt->kdbx_ver == 2 && cur_salt->cipher == 0) {
+				unsigned char dec_buf[32];
 
-			chacha_decrypt_bytes(&ckey, cur_salt->contents, dec_buf, 32, 20);
-			if (!memcmp(dec_buf, cur_salt->expected_bytes, 32)) {
-				cracked[index] = 1;
+				AES_cbc_encrypt(cur_salt->contents, dec_buf, 32,
+				                &akey, iv, AES_DECRYPT);
+				if (!memcmp(dec_buf, cur_salt->expected_bytes, 32)) {
+					cracked[index] = 1;
 #ifdef _OPENMP
 #pragma omp atomic
 #endif
-				any_cracked |= 1;
+					any_cracked |= 1;
+				}
 			}
+			else if (cur_salt->kdbx_ver == 2 && cur_salt->cipher == 2) {
+				unsigned char dec_buf[32];
 
-		}
-		else if (cur_salt->version == 1 && cur_salt->algorithm == 1) { /* KeePass 1.x with Twofish */
-			int crypto_size;
-
-			decrypted_content = mem_alloc(cur_salt->contentsize);
-			crypto_size = Twofish_Decrypt(&tkey, cur_salt->contents,
-			                              decrypted_content,
-			                              cur_salt->contentsize, iv);
-			datasize = crypto_size;  // awesome, right?
-			if (datasize <= cur_salt->contentsize && datasize > 0) {
-				SHA256_Init(&ctx);
-				SHA256_Update(&ctx, decrypted_content, datasize);
-				SHA256_Final(out, &ctx);
-				if (!memcmp(out, cur_salt->contents_hash, 32)) {
+				chacha_decrypt_bytes(&ckey, cur_salt->contents, dec_buf, 32, 20);
+				if (!memcmp(dec_buf, cur_salt->expected_bytes, 32)) {
 					cracked[index] = 1;
 #ifdef _OPENMP
 #pragma omp atomic
 #endif
 					any_cracked |= 1;
 				}
+
+			}
+			else if (cur_salt->kdbx_ver == 1 && cur_salt->cipher == 1) { /* KeePass 1.x with Twofish */
+				int crypto_size;
+
+				decrypted_content = mem_alloc(cur_salt->content_size);
+				crypto_size = Twofish_Decrypt(&tkey, cur_salt->contents,
+				                              decrypted_content,
+				                              cur_salt->content_size, iv);
+				datasize = crypto_size;  // awesome, right?
+				if (datasize <= cur_salt->content_size && datasize > 0) {
+					SHA256_Init(&ctx);
+					SHA256_Update(&ctx, decrypted_content, datasize);
+					SHA256_Final(out, &ctx);
+					if (!memcmp(out, cur_salt->contents_hash, 32)) {
+						cracked[index] = 1;
+#ifdef _OPENMP
+#pragma omp atomic
+#endif
+						any_cracked |= 1;
+					}
+				}
+				MEM_FREE(decrypted_content);
+			} else {
+				// KeePass version 2 with Twofish is TODO. Twofish support under KeePass version 2
+				// requires a third-party plugin. See http://keepass.info/plugins.html for details.
+				error_msg("KeePass v2 w/ Twofish not supported yet\n");
 			}
-			MEM_FREE(decrypted_content);
-		} else {
-			// KeePass version 2 with Twofish is TODO. Twofish support under KeePass version 2
-			// requires a third-party plugin. See http://keepass.info/plugins.html for details.
-			error_msg("KeePass v2 w/ Twofish not supported yet");
 		}
 	}
+	if (failed) {
+#ifdef _OPENMP
+		fprintf(stderr, "Keepass: Error: Argon2 failed in some threads\n");
```

Comment:
Other error messages in this source file (in the allocation functions) start with `Error: KeePass: `, so let's make this one similar.